plugins/module_utils/foreman_helper.py

```diff
@@ -656,8 +659,18 @@ def _entity_spec_helper(spec):
             argument_value['elements'] = 'dict'
             _dummy, argument_value['options'] = _entity_spec_helper(argument_value.pop('entity_spec'))
             entity_value = None
+        elif 'type' in argument_value:
+            entity_value['type'] = argument_value['type']
+
+        if flat_name:
+            entity_value['flat_name'] = flat_name
+            entity_spec[flat_name] = {}
+            if 'type' in argument_value:
+                entity_spec[flat_name]['type'] = argument_value.get('type', 'str')
```

Comment:
isn't the `or 'str'` in the `get()` superfluous, as you already checked that `type` is in `argument_value`?

Reply:
Right.
```suggestion
                entity_spec[flat_name]['type'] = argument_value['type']
```